### PR TITLE
Fixes #183 - Preserve list items.

### LINKED ISSUE #183: Daring Fireball "Colophon" readable version misses list items
blank_skitch_document

### Steps to reproduce:
1. Go to http://daringfireball.net/colophon/
2. Note there is no reader view icon available (at least in Aurora)

### Actual results:
daring_fireball__colophon

### Expected results:
A reader-view capable page. I did notice there is a `<div id="Main"><div class="article">...</div></div>` which i thought may be enough to search and scrape on.

Oddly, I did notice that this page is reader-viewable in Safari (go figure):

daring_fireball__colophon

### PR DESCRIPTION
**Pull request by n1k0**
r=? @gijsk @leibovic

### CODE REVIEW

File: test/test-pages/daringfireball-1/expected-metadata.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "title": "Daring Fireball: Colophon",
+  "byline": null,
```
Comment (n1k0) [inline test/test-pages/daringfireball-1/expected-metadata.json]: Side note: Interesting.

File: Readability.js [hunk 2]
```diff
@@ -1629,17 +1630,17 @@ Readability.prototype = {
...
         var toRemove = false;
         if (img > p && !this._hasAncestorTag(tagsList[i], "figure")) {
           toRemove = true;
         } else if (li > p && tag !== "ul" && tag !== "ol") {
           toRemove = true;
-        } else if ( input > Math.floor(p/3) ) {
+        } else if (input > Math.floor(p/3)) {
           toRemove = true;
-        } else if (contentLength < 25 && (img === 0 || img > 2) ) {
+        } else if (!isList && contentLength < 10 && (img === 0 || img > 2)) {
```
Comment (gijsk) [inline Readability.js]: Why reduce this from 25 to 10?

File: test/test-pages/daringfireball-1/expected-metadata.json (hunk 1 above)
Comment (n1k0) [inline test/test-pages/daringfireball-1/expected-metadata.json]: Actually not, as there's no semantic markup in this page stating who the author is.

File: Readability.js [hunk 3]
```diff
@@ -1598,6 +1598,7 @@ Readability.prototype = {
 
     var tagsList = e.getElementsByTagName(tag);
     var curTagsLength = tagsList.length;
+    var isList = ["ul", "ol"].indexOf(tag) !== -1;
```
Comment (gijsk) [inline Readability.js]: Nit: just use: `tag == "ul" || tag == "ol"`

File: Readability.js (hunk 2 above)
Comment (gijsk) [inline Readability.js]: In fact, why any of the changes here besides the `isList` check?

File: Readability.js [hunk 4]
```diff
@@ -1629,17 +1630,17 @@ Readability.prototype = {
         }
 
         var linkDensity = this._getLinkDensity(tagsList[i]);
-        var contentLength = this._getInnerText(tagsList[i]).length;
+        var contentLength = this._getInnerText(tagsList[i], true).length;
         var toRemove = false;
         if (img > p && !this._hasAncestorTag(tagsList[i], "figure")) {
           toRemove = true;
         } else if (li > p && tag !== "ul" && tag !== "ol") {
```
Comment (gijsk) [inline Readability.js]: It looks like this already has a similar check that could be normalized to use `isList` as well.

File: Readability.js (hunk 3 above)
Comment (n1k0) [inline Readability.js]: The idea was to ease adding more tags in the future, but for lists I think we're already covering everything.

File: Readability.js [hunk 5]
```diff
@@ -1629,17 +1630,17 @@ Readability.prototype = {
         }
 
         var linkDensity = this._getLinkDensity(tagsList[i]);
-        var contentLength = this._getInnerText(tagsList[i]).length;
+        var contentLength = this._getInnerText(tagsList[i], true).length;
```
Comment (gijsk) [inline Readability.js]: Sorry, I thought I mentioned this already... why this change? :-)
